### PR TITLE
Add a KUBERNETES_NODE_* section to build kubelet/kube-proxy for windows

### PR DESCRIPTION
**Pull request by brendandburns**
@pires @ixdy 

Addresses https://github.com/kubernetes/kubernetes/issues/38785 (I hope)

### CODE REVIEW

File: hack/lib/golang.sh [hunk 1]
```diff
@@ -68,6 +81,20 @@ else
...
   readonly KUBE_SERVER_PLATFORMS
 
+  # The node platforms we build for
+  KUBE_NODE_PLATFORMS=(
+    linux/amd64
+    linux/arm
+    linux/arm64
+    linux/s390x
+    windows/amd64
+    windows/386
```
Comment (luxas) [inline hack/lib/golang.sh]: Do we need 386 here? Are you actually using it?

**Review by luxas — COMMENTED** [review]
This is a quick fix indeed and might work, but we should do this properly (in time for v1.6, not a cherrypick candidate) and remove kubelet and the proxy from the server targets and refactor the build/release framework to take the node targets into account

**Comment by brendandburns** [comment]
@k8s-bot node e2e test this

**Comment by brendandburns** [comment]
@k8s-bot cri node e2e test this

**Comment by ixdy** [comment]
@k8s-bot build this

File: hack/lib/golang.sh (hunk 1 above)
Comment (ixdy) [inline hack/lib/golang.sh]: probably not, though apparently there are still `windows/386` users out there. x-ref https://github.com/kubernetes/kubernetes/issues/28632

Comment (brendandburns) [inline hack/lib/golang.sh]: Removed. Windows 2016 is required for Windows containers, and it's not build for 386 afaik know.

**Comment by brendandburns** [comment]
@ixdy @luxas I need to add the relevant build-node-tarballs stuff, then I think this should be ready to go.

I'll add it this evening.

**Comment by brendandburns** [comment]
@k8s-bot build this

**Comment by brendandburns** [comment]
@k8s-bot build this

**Comment by brendandburns** [comment]
@k8s-bot build this

**Comment by brendandburns** [comment]
@ixdy @luxas This appears to work correctly and packages a new node tarball.

@luxas I don't want to remove them from the server tarball (yet) because that would likely break numerous release scripts.

Eventually (probably not even in 1.6, but maybe 1.7) we can remove them from the server tarball...

Please take another look.

Thanks!
--brendan

**Comment by luxas** [comment]
LGTM

Yes, I didn't mean it should be done now, but maybe for v1.7 we should remove kubelet and the proxy from the server targets

@ixdy can apply the label if he also thinks it's ok

**Comment by brendandburns** [comment]
Ping to @ixdy for approval on this.

Thanks!

**Review by pires — APPROVED** [review]
This works! Thanks @brendandburns.

**Comment by brendandburns** [comment]
@ixdy are you still around?  I'm tempted to LGTM this since both @pires and @luxas approved...

@fejta @spxtr for LGTM approval.

**Comment by ixdy** [comment]
yes, sorry, I will take a look today.

File: build/lib/release.sh [hunk 2]
```diff
@@ -148,6 +149,50 @@ function kube::release::package_client_tarballs() {
...
+    if [[ "${platform%/*}" == "windows" ]]; then
+      node_bins=("${KUBE_NODE_BINARIES_WIN[@]}")
+    fi
+    # This fancy expression will expand to prepend a path
+    # (${LOCAL_OUTPUT_BINPATH}/${platform}/) to every item in the
+    # KUBE_NODE_BINARIES array.
+    cp "${node_bins[@]/#/${LOCAL_OUTPUT_BINPATH}/${platform}/}" \
+      "${release_stage}/node/bin/"
+
+    # TODO: Docker images here
```
Comment (ixdy) [inline build/lib/release.sh]: open an issue for this?

**Review by ixdy — APPROVED** [review]
basically LG. you'll probably need to update https://github.com/kubernetes/release/blob/master/lib/releaselib.sh#L480-L491 too.

**Comment by ixdy** [comment]
/lgtm

I'd like to see an issue filed to possibly remove these from the server tarball, so we reduce duplication.

**Pull request merged**